### PR TITLE
add more tests for CaliperEngine

### PR DESCRIPTION
**Pull request by tunedev**
## Checklist
 - [ ]  A link to the issue/user story that the pull request relates to
 - [ ]  How to recreate the problem without the fix
 - [ ]  Design of the fix
 - [ ]  How to prove that the fix works
 - [ ]  Automated tests that prove the fix keeps on working
 - [ ]  Documentation - any JSDoc, website, or Stackoverflow answers?

## Issue/User story

Partially addresses #1557

## Steps to Reproduce

1.
2.
3.
4.

## Existing issues

- [ ] [Stack Overflow issues](http://stackoverflow.com/tags/hyperledger-caliper)
- [x] [GitHub Issues](https://github.com/hyperledger/caliper/issues)
- [ ] [Discord history](https://discord.com/channels/905194001349627914/941417677778473031)

## Validation of the fix

After this PR: Running the tests in caliper-core the listed %stmts in the code coverage report should tally with the below listed

- caliper-core/lib/manager/caliper-engine.js: 100

Before the PR: Running the tests in caliper-core the listed %stmts in the code coverage report the below listed was what was there before

- caliper-core/lib/manager/caliper-engine.js: 65.11

### CODE REVIEW

File: packages/caliper-core/test/manager/caliper-engine.js [hunk 1]
```diff
@@ -38,125 +56,472 @@ mockery.enable({
...
 
-            it('should handle errors during end command execution', function() {
-                // TODO: Implement test
+            it('should handle errors during end command execution', async function() {
+                stubbedCaliperUtils.getFlowOptions.returns({
+                    performStart: false,
+                    performInit: false,
+                    performInstall: false,
+                    performTest: false,
+                    performEnd: true
```
Comment (davidkel) [inline packages/caliper-core/test/manager/caliper-engine.js:435]: Same comment as before

File: packages/caliper-core/test/manager/caliper-engine.js [hunk 2]
```diff
@@ -38,125 +56,472 @@ mockery.enable({
...
+                    performStart: false
+                });
+
+                mockAdapter.init.rejects(new Error('Init failed'));
+                stubbedCaliperUtils.execAsync.resolves();
+
+                const networkConfig = { caliper: { command: { end: 'cleanup.sh' } } };
+                const adapterFactory = sinon.stub().resolves(mockAdapter);
+                const engine = new CaliperEngine({}, networkConfig, adapterFactory);
+
```
Comment (davidkel) [inline packages/caliper-core/test/manager/caliper-engine.js:461]: Can remove blank line

**Review by davidkel — CHANGES_REQUESTED** [review]
@tunedev Only a minor issue about how we setup the flow options and then just need to set the specific flow option to true that is being tested, plus cleanup of the whitespace. But all good apart from that, thanks for the PR (sorry it took so long to review)
